apps/docs/app/features/routes/ressurser/design-tokens/ColorTokens.tsx

```diff
@@ -139,30 +163,37 @@ const ColorToken = ({ token, ...rest }: ColorTokenProps) => {
   );
 };
 
-/**
- * Accepts a color value, and returns the palette name of that color
- */
-const getPaletteName = (value: string) => {
-  const normalizedValue = value.toLowerCase();
-  const token = Object.values(tokens.color.palette)
-    .flatMap((scale) => (scale.value ? [scale] : Object.values(scale)))
-    .find(
-      (token) =>
-        token.original.value === normalizedValue ||
-        token.value === normalizedValue
-    );
+/** Returns the relevant display information about a token */
+const useTokenInfo = (colorValue: string) => {
+  const normalizedColorValue = colorValue?.toLowerCase() ?? "";
+  const aliasName = getAliasName(normalizedColorValue);
+  const paletteName = getPaletteName(normalizedColorValue);
+  return {
+    aliasName: aliasName ?? paletteName,
+    paletteName: paletteName ?? aliasName,
+    colorValue: normalizedColorValue.startsWith("#")
+      ? normalizedColorValue.toUpperCase()
+      : normalizedColorValue,
+  };
+};
 
-  return token
-    ? `${token.attributes.item} ${token.attributes.subitem ?? ""}`
-    : value;
+const getAliasName = (colorValue: string) => {
+  const entry = Object.entries(tokens.color.alias).find(
+    ([_, value]) => colorValue === value
+  );
+  return entry ? toTitleCase(entry[0]) : null;
 };
 
 /**
- * Accepts a color value, and returns the alias name of that color - if there is an alias for that color.
+ * Accepts a color value, and returns the palette name of that color
  */
-const getAliasName = (value: string) => {
-  const token = Object.values(tokens.color.alias).find(
-    (token) => token.value.toLowerCase() === value.toLowerCase()
-  );
-  return token ? token.attributes.item : null;
+const getPaletteName = (colorValue: string) => {
```

Comment:
Åpen for forbedringsforslag på denne koden - den føles litt tung ut. Men den ser ut til å kjøre ganske fort.